mobile/library/common/http/client.cc

```diff
@@ -406,54 +412,77 @@ void Client::DirectStream::saveLatestStreamIntel() {
 }
 
 void Client::DirectStream::saveFinalStreamIntel() {
+  if (!parent_.getStream(stream_handle_, ALLOW_ONLY_FOR_OPEN_STREAMS)) {
```

Comment:
reordering because we now have the envoy_bug in requestDecoder() and checking getStream first avoids failing that check.